### PR TITLE
fix(icons-react): fix exports in individual icon TypeScript declarations

### PR DESCRIPTION
**Pull request by lewandom**
Fixes the problem raised in https://github.com/carbon-design-system/carbon/pull/14714#issuecomment-1777878629: improper typescript type exports in individual icons TypeScript declaration file (.d.ts).

#### Changelog

**Changed**

- changed variable export to default export in individual icons TypeScript declaration file (.d.ts) to match the actual JS code

#### Testing / Reviewing

Tested in TypeScript project by linking the patched Carbon version. Can import individual icon using default import.

### CODE REVIEW

**Comment by lewandom** [comment]
@metonym FYI.

File: packages/icon-build-helpers/src/builders/react/next/typescript.js [hunk 1]
```diff
@@ -46,9 +46,8 @@ async function writeModuleTypes(modules, outDir) {
       templates.banner +
       '\n' +
       "import type { CarbonIconType } from './CarbonIcon';\n" +
-      'export const ' +
-      m.name +
-      ': CarbonIconType;\n';
+      `const ${m.name}: CarbonIconType;\n` +
+      `export default ${m.name}\n`;
```
Comment (metonym) [inline packages/icon-build-helpers/src/builders/react/next/typescript.js:50]: FYI, I see this locally when copying/pasting this change to my local environment:

Screenshot 2023-10-26 at 8 30 35 AM

Comment (lewandom) [reply packages/icon-build-helpers/src/builders/react/next/typescript.js:50]: @metonym thanks for spotting! I added the missing declare keyword and it seems to be better now. Please confirm if this also helps in your setup.

Comment (metonym) [reply packages/icon-build-helpers/src/builders/react/next/typescript.js:50]: LGTM

**Review by metonym — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**